mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -113,6 +113,12 @@ struct GPUInlinerInterface : public DialectInlinerInterface {
                        BlockAndValueMapping &) const final {
     return true;
   }
+
+  /// All functions can be inlined.
+  bool isLegalToInline(Region *, Region *, bool,
```

Comment:
Should we update the comment to clarify that all functions can be inlined into a gpu.func region, and not gpu.func can always be inlined?

Reply:
As written the member function allows inlining any region into a gpu.func region, unless the region contains operations deemed to be not inlinable by the dialect they belong too.
  
You comment made me think this is too permissive and I will change the code to avoid permitting a gpu.func to be inlined into another gpu.func.  Will also expand the test to check for this.